x-pack/plugins/lens/public/shared_components/legend_settings_popover.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiButtonGroup, EuiSpacer, EuiSwitch, EuiSwitchEvent } from '@elastic/eui';
+import { Position } from '@elastic/charts';
+import { ToolbarPopover } from '../shared_components';
+
+interface LegendSettingsPopoverProps {
+  /**
+   * Determines the legend display options
+   */
+  legendOptions: Array<{ id: string; value: 'auto' | 'show' | 'hide' | 'default'; label: string }>;
+  /**
+   * Determines the legend mode
+   */
+  mode: 'default' | 'show' | 'hide' | 'auto';
+  /**
+   * Callback on display option change
+   */
+  onDisplayChange: (id: string) => void;
+  /**
+   * Sets the legend position
+   */
+  position?: Position;
+  /**
+   * Callback on position option change
+   */
+  onPositionChange: (id: string) => void;
+  /**
+   * If true, nested legend switch is rendered
+   */
+  renderNestedLegendSwitch?: boolean;
+  /**
+   * nested legend switch status
+   */
+  nestedLegend?: boolean;
+  /**
+   * Callback on nested switch status change
+   */
+  onNestedLegendChange?: (event: EuiSwitchEvent) => void;
+}
+
+const toggleButtonsIcons = [
+  {
+    id: Position.Bottom,
+    label: 'Bottom',
+    iconType: 'arrowDown',
+  },
+  {
+    id: Position.Left,
+    label: 'Left',
+    iconType: 'arrowLeft',
+  },
+  {
+    id: Position.Right,
+    label: 'Right',
+    iconType: 'arrowRight',
+  },
+  {
+    id: Position.Top,
+    label: 'Top',
+    iconType: 'arrowUp',
+  },
+];
+
+export const LegendSettingsPopover: React.FunctionComponent<LegendSettingsPopoverProps> = ({
+  legendOptions,
+  mode,
+  onDisplayChange,
+  position,
+  onPositionChange,
+  renderNestedLegendSwitch = false,
+  nestedLegend = false,
+  onNestedLegendChange = () => {},
+}) => {
+  return (
+    <ToolbarPopover
+      title={i18n.translate('xpack.lens.xyChart.legendLabel', {
```

Comment:
Nitpick: You could change the i18n ID to represent that it's shared, or leave it.